src/nc/arch/x86/X86InstructionAnalyzer.cpp

```diff
@@ -133,32 +133,87 @@ class X86InstructionAnalyzerImpl {
         /* Describing semantics */
         switch (ud_obj_.mnemonic) {
             case UD_Iadc: {
+#define adc_bitness(bitness) if(operand(0).size() == bitness)\
+                _[\
+                    regizter(X86Registers::tmp##bitness ()) ^= operand(0) + operand(1) + zero_extend(cf),\
+                    cf ^= unsigned_(regizter(X86Registers::tmp##bitness ())) < unsigned_(operand(0)),\
+                    operand(0) ^= regizter(X86Registers::tmp##bitness ()),\
+                    pf ^= intrinsic(),\
+                    zf ^= operand(0) == constant(0),\
+                    sf ^= signed_(operand(0)) < constant(0),\
+                    of ^= intrinsic(),\
+                    af ^= intrinsic(),\
+                    less ^= ~(sf == of),\
+                    less_or_equal ^= less | zf,\
+                    below_or_equal ^= cf | zf\
+                ]; else{}
+
+                adc_bitness(8);
+                adc_bitness(16);
+                adc_bitness(32);
+                adc_bitness(64);
+#undef adc_bitness
+                break;
+            }
+            case UD_Iadd: {
+#define add_bitness(bitness) if(operand(0).size() == bitness)\
```

Comment:
You do not need a macro here (and in adc too).
There is `temporary()` function, specifically for the case when you want a temporary register of a size unknown at compile-time.